src/ansi-c/c_typecheck_base.cpp

```diff
@@ -822,7 +845,9 @@ void c_typecheck_baset::typecheck_declaration(
         for(auto &expr : code_type.requires_contract())
         {
           typecheck_spec_function_pointer_obeys_contract(expr);
-          check_history_expr(expr);
+          std::string clause_type = "function pointer preconditions";
+          check_history_expr(expr, clause_type);
+          check_return_value(expr, clause_type);
```

Comment:
Do we always do both of these? Or does one always entail the other? If so, this ought to be refactored into a single call.